core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -112,6 +112,38 @@ public void deprecated(String msg, Object... params) {
         deprecated(THREAD_CONTEXT, msg, params);
     }
 
+    /*
+     * RFC7234 specifies the warning format as warn-code <space> warn-agent <space> "warn-text" [<space> "warn-date"]. Here, warn-code is a
+     * three-digit number with various standard warn codes specified. The warn code 299 is apt for our purposes as it represents a
```

Comment:
To indicate that this is a persistent warning as opposed to a transient warning (e.g., a warning that could expire after time elapses).